### PR TITLE
[Unit Testing] - DataRepository, RoomsDao #1

### PR DESCRIPTION
**Pull request by Dairai**
Initial Unit tests for DataRepository and RoomsDao classes

### CODE REVIEW

File: src/com/comeet/utilities/Validator.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.comeet.utilities;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Validator {
+    
+    private static final Pattern VALID_EMAIL_ADDRESS_REGEX =
```
Comment (kb000) [inline src/com/comeet/utilities/Validator.java:8]: Email addresses are very difficult to validate by regex.  
http://stackoverflow.com/questions/201323/using-a-regular-expression-to-validate-an-email-address
What standard did you use here? Please cite it.

Comment (kb000) [reply src/com/comeet/utilities/Validator.java:8]: now, *exchange* addresses might be more restrictive. but I'm not going to go digging up documentation on this.

File: test/com/comeet/RoomsDaoTest.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    @Mock
+    ApiLogger log;
+    
+    @Before
+    public void setUp() throws IOException {
+        //ApiLogger.setup();
+    }
+    
+    //GetCriteria tests
+    //Note: This method calls a data repository class method therefore
```
Comment (kb000) [inline test/com/comeet/RoomsDaoTest.java:39]: To make these tests work independently of our db, you should really be injecting a mock `sqlConnection` with known data.

File: test/com/comeet/RoomsDaoTest.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    @Test(expected=IllegalArgumentException.class)
+    public void getBuildingRoomsISODateTest() throws Exception {
+        String start = DateTime.now().toDateTimeISO().toString();
+        String end = DateTime.now().plusDays(7).toDateTimeISO().toString();
+        
+        Assert.assertSame(Exception.class, 
+                        dao.getBuildingRooms("Bldg_CambMaOneStorySt@meetl.ink",start,end)
+                        .getClass());
+        
+        Assert.assertSame(Exception.class,
```
Comment (kb000) [inline test/com/comeet/RoomsDaoTest.java:106]: This works?
I wouldn't think code would execute past the first exception-causing statement, above.
you can verify the execptions by breaking this up into three separate test methods (with 3 expected exceptions), or by using

    try {
        somethingBad();
        Assert.fail("Expected exception to be thrown");
    } catch (IllegalArgumentException e) {
        Assert.true(e.getMessage().contains("myArgument");
    }`

**Review by kb000 — APPROVED** [review]
I'm glad we have some unit tests rather than none,  but I'm sad that they're the weakest kind of validation (class matching...), instead of verifying unit functional requirements.

File: test/com/comeet/RoomsDaoTest.java (hunk 2 above)
Comment (Dairai) [reply test/com/comeet/RoomsDaoTest.java:39]: Good to know. I will read up on this and update the test cases

File: src/com/comeet/utilities/Validator.java (hunk 1 above)
Comment (Dairai) [reply src/com/comeet/utilities/Validator.java:8]: I had used a regex pattern based off of common email address patterns. I did some follow up and found a pattern that conforms to RFC 822 (http://www.ietf.org/rfc/rfc0822.txt?number=822) that I will include in my next pull request (http://www.ex-parrot.com/pdw/Mail-RFC822-Address.html)

**Pull request merged**